apps/epic-web/src/lib/products.ts

```diff
@@ -57,3 +58,106 @@ export async function getProduct(productId: string): Promise<Product> {
+      status: product.state === 'active' ? 1 : 0,
+      quantityAvailable: -1,
+      // TODO: upgradableTo, coupons
+      ...(product.price && {
+        prices: {
+          connectOrCreate: {
+            where: {
+              id: product.price.priceId,
+            },
+            create: {
+              id: product.price.priceId,
+              unitAmount: product.price.amount,
+              merchantPrices: {
+                connectOrCreate: {
+                  where: {
+                    id: product.merchantPrice.merchantPriceId,
+                  },
+                  create: {
+                    id: product.merchantPrice.merchantPriceId,
+                    identifier: product.merchantPrice.identifier,
+                    merchantProductId: product.productId,
+                    merchantAccountId: product.merchantPrice.merchantAccountId,
+                  },
+                },
+              },
+            },
+          },
+        },
+      }),
+      ...(product.merchantProduct && {
+        merchantProducts: {
+          connectOrCreate: {
+            where: {
+              id: product.merchantProduct.merchantProductId,
+            },
+            create: {
+              id: product.merchantProduct.merchantProductId,
+              identifier: product.merchantProduct.identifier,
+              merchantAccountId: product.merchantProduct.merchantAccountId,
+            },
+          },
+        },
+      }),
+    },
+    update: {
+      name: product.title,
+      status: product.state === 'active' ? 1 : 0,
+      quantityAvailable: -1,
+      ...(product.price && {
+        prices: {
+          connectOrCreate: {
+            where: {
+              id: product.price.priceId,
+            },
+            create: {
+              id: product.price.priceId,
+              unitAmount: product.price.amount,
+              merchantPrices: {
+                connectOrCreate: {
+                  where: {
+                    id: product.merchantPrice.merchantPriceId,
+                  },
+                  create: {
+                    id: product.merchantPrice.merchantPriceId,
+                    identifier: product.merchantPrice.identifier,
+                    merchantProductId: product.productId,
+                    merchantAccountId: product.merchantPrice.merchantAccountId,
+                  },
+                },
+              },
+            },
+          },
+        },
+      }),
+      ...(product.merchantProduct && {
+        merchantProducts: {
+          connectOrCreate: {
+            where: {
+              id: product.merchantProduct.merchantProductId,
+            },
+            create: {
+              id: product.merchantProduct.merchantProductId,
+              identifier: product.merchantProduct.identifier,
+              merchantAccountId: product.merchantProduct.merchantAccountId,
+            },
+          },
+        },
+      }),
+    },
+  })
+  return newProduct
+}
```

Comment:
not super stoked on this and think in particular we need to reconsider the `any` for `product`

any ideas @jbranchaud? this could be a followup